### ISSUE TITLE
Enforce file extension for import declarations

### ISSUE THREAD

**Issue by viktor-ku** [eslint/eslint#11894]
**Please describe what the rule should do:**

It should enforce file extension usage in the [`FromClause`](https://www.ecma-international.org/ecma-262/9.0/index.html#prod-FromClause) of the [`ImportDeclaration`](https://www.ecma-international.org/ecma-262/9.0/index.html#prod-ImportDeclaration).

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
// relative path to a directory or a file without extension
import { Counter } from '../components/counter'

// aliased path to a directory or a file without extension
import { Counter } from '@/components/counter'

// absolute path to a directory or a file without extension
import { Counter } from '/home/user/code/app/components/counter'
```

The only legal exception would be to use the module name

```js
import _ from 'lodash'
import get from 'lodash/get'
```

Why? 

1. [Import maps](https://wicg.github.io/import-maps/) currently in a draft stage, but if it becomes the standard then it will be the only option to actually use something like this in the native `module` type scripts.
2. To support currently existing applications written with webpack (or similar) that are using `node_modules` to import modules.

**Why should this rule be included in ESLint (instead of a plugin)?**

The main reason is:

- It is 100% required rule to write native browser module scripts

As far as I know it is becoming essential in node.js as well when using `.mjs` or module type.

It should set a good practice as well as help migrating existing codebases to a new standard.

**Are you willing to submit a pull request to implement this rule?**

Yes, I am
